webapp/src/components/link_fallback.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { useEffect, useState } from 'react';
+import {Link} from '@mui/material';
+
+
+interface Props {
+    fallback: boolean
+    link?: string;
+    children: React.ReactNode;
+}
+
+const LinkFallback = ({fallback, link, children}: Props) => {
+    const [showLink, setShowLink] = useState(false)
```

Comment:
### The context behind the actual reason
Before the state introduction, i simply rendered place holder text if the content element was not the last and the content itself otherwise. It created a no so good UX when a new content was added at the end of the conversation (like a new video). In this case previously last video was switched to link representation & this sudden switch was a bit off.
So i tried to avoid sudden component switches by keeping the view that was rendered the first time. 
Meaning that:
- Videos that were last & actually shown on video player will stay to be shown as videos even after moving up the list (when some other video content is added at the end of the chat)
- All the old video/IDE-s will be rendered as links when user first enters the chat

Of course if this is no needed or i'm missing anything, i'm more then open to suggestions/changes

Reply:
"sudden component switch" is the intended UX. Otherwise, the page becomes "heavy", with lots of youtube players rendering

Reply:
Got that, will change it.
Thanks for response